compiler/circle-mpqsolver/src/core/DumpingHooks.h

```diff
@@ -68,14 +68,15 @@ class DumpingHooks final : public QuantizerHook, public SolverHooks
    * @brief called at the end of iterative search
    */
   virtual void on_end_solver(const LayerParams &layers, const std::string &def_dtype,
-                             float qerror) override;
+                             const std::string *def_granularity, const float *qerror) override;
```

Comment:
> If _ctx is given, def_dtype and def_granularity would be unnecessary arguments?

@jinevening 
For `bisection` def_dtype is needed, because we can quantize to `int16` the whole network.
As for `def_granularity` - you are right, we don't need. I'll remove it.

Reply:
@jinevening 
done.

Reply:
@jinevening 
Please take another look. `BisectionSolver` still uses explicit "uint8"/"int16". Do i have to change `BisectionSolver`, to use only context?

Reply:
Looks okay to me. Let's proceed.